src/tree_schema.h

```diff
@@ -111,14 +111,101 @@ extern "C" {
  * @param ELEM Iterator intended for use in the block.
  */
 
-#ifdef __cplusplus
-#define TYPES_COMPATIBLE(type1, type2) typeid(*(type1)) == typeid(type2)
-#elif defined(__GNUC__) || defined(__clang__)
-#define TYPES_COMPATIBLE(type1, type2) __builtin_types_compatible_p(__typeof__(*(type1)), type2)
-#else
-#define TYPES_COMPATIBLE(type1, type2) _Generic(*(type1), type2: 1, default: 0)
-#endif
+#if defined(_WINDOWS) && defined(_MSC_VER) && _MSC_VER < 1927
+  // MSVC does not support compile-time introspection via _Generic before VS2019 version 16.7
+  // https://developercommunity.visualstudio.com/idea/443508/msvc-2017-does-not-support-generic-type-generic-ma.html
```

Comment:
isn't there really a support for `__builtin_types_compatible_p` or something like that? We actually does not need whole the Generic mechanism, we just need to have basic knowledge about the type to decide how to access the structures.

Reply:
Unfortunately __builtin_types_compatible_p  is GCC-specific. Surprisingly MSVC provides does not provide any equivalent.